WooCommerce/Classes/ViewRelated/Blaze/TargetDevices/BlazeTargetDevicePickerView.swift

```diff
@@ -0,0 +1,57 @@
+import SwiftUI
+import struct Yosemite.BlazeTargetDevice
+
+/// View for picking target devices for a Blaze campaign
+struct BlazeTargetDevicePickerView: View {
+
+    @ObservedObject private var viewModel: BlazeTargetDevicePickerViewModel
+
+    private let selectedDevices: Set<BlazeTargetDevice>?
+    private let onDismiss: () -> Void
+
+    init(viewModel: BlazeTargetDevicePickerViewModel,
+         selectedDevices: Set<BlazeTargetDevice>?,
+         onDismiss: @escaping () -> Void) {
+        self.viewModel = viewModel
+        self.selectedDevices = selectedDevices
+        self.onDismiss = onDismiss
+    }
+
+    var body: some View {
+        MultiSelectionList(title: Localization.title,
+                           allOptionsTitle: Localization.allTitle,
+                           contents: viewModel.devices,
+                           contentKeyPath: \.name,
+                           selectedItems: selectedDevices,
+                           onDismiss: onDismiss,
+                           onCompletion: { selectedDevices in
+                                viewModel.confirmSelection(selectedDevices)
+                                onDismiss()
+                            })
+        .task {
+            await viewModel.syncDevices()
```

Comment:
❓ Should we consider showing a loading indicator when we sync the devices? Will there will be a delay the first time we load these values from the server?

Reply:
Adding loading indicator will require refactoring the `MultiSelectionList` to extract the `NavigationView` and toolbar. I'll add a separate task for this.